src/sparql/QuantityValuePatternBuilder.ts

```diff
@@ -0,0 +1,259 @@
+			statementVariable,
+			[
+				rdfNamespaces.psn + propertyId,
+				rdfNamespaces.wikibase + 'quantityAmount',
+			],
+			statementQuantityVariable,
+		);
+		const statementToUnitTyple = SyntaxBuilder.buildPathTriple(
+			statementVariable,
+			[
+				rdfNamespaces.psn + propertyId,
+				rdfNamespaces.wikibase + 'quantityUnit',
+			],
+			coherentUnitVariable,
+		);
+		const entityQuantityGroup = SyntaxBuilder.buildBgpPattern(
+			[
+				entityToStatementTriple,
+				statementToAmountTriple,
+				statementToUnitTyple,
+			],
+		);
+		patterns.push( entityQuantityGroup );
+
+		const operator = this.getFilterOperatorFromRelation( propertyValueRelation );
+		if ( operator !== null ) {
+			const valueFilterPattern = SyntaxBuilder.buildOperatorFilterPattern(
+				statementQuantityVariable,
+				operator,
+				coherentUserQuantityVariable,
+			);
+			patterns.push( valueFilterPattern );
+		}
+
+		const referenceFilterPattern = this.tripleBuilder.buildReferenceFilterPattern(
+			referenceRelation,
+			statementVariable,
+		);
+		if ( referenceFilterPattern !== null ) {
+			patterns.push( referenceFilterPattern );
+		}
+
+		return patterns;
+	}
+
+	private buildPatternsForUnitlessValue(
+		propertyId: string,
+		numericValue: number,
+		conditionIndex: number,
+		referenceRelation: ReferenceRelation,
+		propertyValueRelation: PropertyValueRelation,
+	): Pattern[] {
+		const patterns: Pattern[] = [];
+		const numericQuantityVariable = SyntaxBuilder.buildVariableTermFromName( 'numericQuantity' );
+		const statementVariable = SyntaxBuilder.buildVariableTermFromName( 'statement' + conditionIndex );
+		const bgp = SyntaxBuilder.buildBgpPattern( [
+			SyntaxBuilder.buildSimpleTriple(
+				{
+					termType: 'Variable',
+					value: 'item',
+				},
+				rdfNamespaces.p + propertyId,
+				statementVariable,
+			),
+			SyntaxBuilder.buildPathTriple(
+				statementVariable,
+				[
+					rdfNamespaces.psv + propertyId,
+					rdfNamespaces.wikibase + 'quantityAmount',
+				],
+				numericQuantityVariable,
+			),
+		] );
+		patterns.push( bgp );
+
+		const operator = this.getFilterOperatorFromRelation( propertyValueRelation );
+		if ( operator !== null ) {
+			const valueFilterPattern = SyntaxBuilder.buildOperatorFilterPattern(
+				numericQuantityVariable,
+				operator,
+				SyntaxBuilder.buildLiteralTermForDecimalNumber( numericValue ),
+			);
+			patterns.push( valueFilterPattern );
+		}
+
+		const referenceFilterPattern = this.tripleBuilder.buildReferenceFilterPattern(
+			referenceRelation,
+			statementVariable,
+		);
+		if ( referenceFilterPattern !== null ) {
+			patterns.push( referenceFilterPattern );
+		}
+
+		return patterns;
+	}
+
+	private isQuantityValue( value: string | UnitValue ): value is UnitValue {
+		return typeof value === 'object' &&
+			( ( value as UnitValue ).unit === null || typeof ( value as UnitValue ).unit === 'string' );
+	}
+
+	private getFilterOperatorFromRelation( propertyValueRelation: PropertyValueRelation ): string | null {
```

Comment:
This probably can be moved to a central place but can happen later.

Reply:
Let's wait for `DateValuePatternBuilder` to land to see what options we're really dealing with.